accounts-db/src/accounts_db.rs

```diff
@@ -8948,30 +8932,33 @@ impl AccountsDb {
 
     fn generate_index_for_slot(
         &self,
-        accounts_map: GenerateIndexAccountsMap<'_>,
+        storage: &Arc<AccountStorageEntry>,
         slot: Slot,
         store_id: AppendVecId,
         rent_collector: &RentCollector,
         storage_info: &StorageSizeAndCountMap,
+        approx_stored_count: usize,
     ) -> SlotIndexGenerationInfo {
-        if accounts_map.is_empty() {
+        let mut accounts = storage.accounts.account_iter();
+        if accounts.next().is_none() {
```

Comment:
not sure. since we are at startup, the fields behind is_empty may not be initialized properly yet. To protect the rest of the fn against empty, this is the most reliable way.